ngraph/core/src/op/gelu.cpp

```diff
@@ -93,8 +95,109 @@ void op::Gelu::pre_validate_and_infer_types()
+        return enum_names;
     }
+
+    constexpr DiscreteTypeInfo AttributeAdapter<op::GeluApproximationMode>::type_info;
+
+    std::ostream& op::operator<<(std::ostream& s, const op::GeluApproximationMode& type)
+    {
+        return s << as_string(type);
+    }
+} // namespace ngraph
+
+NGRAPH_RTTI_DEFINITION(op::v7::Gelu, "Gelu", 7);
+
+op::v7::Gelu::Gelu(const Output<Node>& data, GeluApproximationMode mode)
+    : UnaryElementwiseArithmetic(data)
+    , m_approximation_mode(mode)
+{
+    constructor_validate_and_infer_types();
+}
+
+bool op::v7::Gelu::visit_attributes(AttributeVisitor& visitor)
+{
+    NGRAPH_OP_SCOPE(v7_Gelu_visit_attributes);
+    visitor.on_attribute("approximation_mode", m_approximation_mode);
+    return true;
+}
+
+shared_ptr<Node> op::v7::Gelu::clone_with_new_inputs(const OutputVector& new_args) const
+{
+    NGRAPH_OP_SCOPE(v7_Gelu_clone_with_new_inputs);
+    if (new_args.size() != 1)
+    {
+        throw ngraph_error("Incorrect number of new arguments");
+    }
+    return make_shared<op::v7::Gelu>(new_args.at(0), m_approximation_mode);
+}
+
+void op::v7::Gelu::validate_and_infer_types()
+{
+    element::Type input_element_type = get_input_element_type(0);
+    PartialShape input_pshape = get_input_partial_shape(0);
+
+    NODE_VALIDATION_CHECK(this,
+                          input_element_type.is_dynamic() || input_element_type.is_real(),
+                          "Argument element type must be f16, bf16, f32, f64 or dynamic (got ",
+                          input_element_type,
+                          ").");
+
+    set_output_type(0, input_element_type, input_pshape);
+}
+
+op::GeluApproximationMode op::v7::Gelu::get_approximation_mode() const
+{
+    return m_approximation_mode;
+}
+
+namespace gelu
+{
+    template <element::Type_t ET>
+    inline bool evaluate(const HostTensorPtr& arg0,
+                         const HostTensorPtr& out,
+                         op::GeluApproximationMode mode,
+                         const size_t count)
+    {
+        using T = typename element_type_traits<ET>::value_type;
+        runtime::reference::gelu<T>(arg0->get_data_ptr<ET>(), out->get_data_ptr<ET>(), mode, count);
+        return true;
+    }
+
+    bool evaluate_gelu(const HostTensorPtr& arg0,
+                       const HostTensorPtr& out,
+                       op::GeluApproximationMode mode,
+                       const size_t count)
+    {
+        bool rc = true;
+        out->set_unary(arg0);
+
+        switch (arg0->get_element_type())
+        {
+            NGRAPH_TYPE_CASE(evaluate_gelu, f16, arg0, out, mode, count);
+            NGRAPH_TYPE_CASE(evaluate_gelu, f32, arg0, out, mode, count);
+        default: rc = false; break;
+        }
+        return rc;
+    }
+}
+
+bool op::v7::Gelu::evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs) const
+{
+    NGRAPH_OP_SCOPE(v7_Gelu_evaluate);
+    return gelu::evaluate_gelu(
```

Comment:
Do we have real examples for gelu operation on the constant folding path?

Reply:
No, but the operation is simple and the binary footprint increase is negligible.